[CDAP-2066] Added DB Batch Sink

bdmogal opened the pull request:
Adds a ``DBSink`` that:
1. Reads a ``StructuredRecord``
2. Makes a JDBC call to the destination table to figure out the schema of the destination table as ``ResultSetMetadata``. This is needed because Database Drivers do not reliably return the schema from a ``PreparedStatement``.
3. Constructs a ``DBRecord`` using the ``StructuredRecord`` and ``ResultSetMetadata`` from 1 and 2 above.
4. Sends the ``DBRecord`` a custom ``ETLDBOutputFormat``, which is needed to load the driver class properly, just like we do in ``ETLDBInputFormat``.

- [x] Tested most SQL column types.
- [x] Tested on a local MySQL.
- [x] Tested a CDAP Table to MySQL table ETL pipeline on a cluster.
- [ ] The unit test for this fails, because when ``DBOutputFormat`` calls ``PreparedStatement.executeBatch()``, HSQLDB throws an exception saying *Statement is not in batch mode.* I'm trying to debug this a little more, but if it does not work, we may have to rely instead on an audi test that sets up a local mysql for this.

Jira: [CDAP-2066](https://issues.cask.co/browse/CDAP-2066)
Build: https://builds.cask.co/browse/CDAP-DT26

albertshau commented on cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/batch/sinks/DBSink.java:
```diff
@@ -0,0 +1,171 @@
+                   false));
+    configurer.addProperty(
+      new Property(Properties.DB.PASSWORD,
+                   "Password to be use to connect to the specified database. " +
```
typo: Password to use

bdmogal replied:
Fixed

albertshau commented on cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/common/ETLDBOutputFormat.java:
```diff
@@ -0,0 +1,82 @@
+ *
+ * {@inheritDoc}
+ */
+public class ETLDBOutputFormat<K  extends DBWritable, V>  extends DBOutputFormat<K, V> {
```
remove extra space between K and extends

bdmogal replied:
Fixed

albertshau commented:
minor comments, otherwise lgtm

bdmogal commented:
Fixed nits. Rebased with develop. Tested a complete local run (unit tests with local mysql). 

Latest Build Passed - http://builds.cask.co/browse/CDAP-DT26-5. Merging.
